standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/PersistenceManagerProvider.java

```diff
@@ -653,7 +653,6 @@ private static <T> T retry(Supplier<T> s) {
             LOG.warn("Exception retry limit reached, not retrying any longer.", e);
           } else {
             LOG.debug("Non-retriable exception.", e);
-            break;
```

Comment:
Why remove the `break`?  When encountering an un-retriable exception, shouldn't we immediately exit the loop?

Reply:
BTW, Should we place `ex = e` before the` if else` block to ensure it executes even if a break is encountered?
```
@@ -649,13 +649,13 @@ private static <T> T retry(Supplier<T> s) {
           // If we're here, we'll proceed down the next while loop iteration.
         } else {
           // we've reached our limit, throw the last one.
+          ex = e;
           if (retriable) {
             LOG.warn("Exception retry limit reached, not retrying any longer.", e);
           } else {
             LOG.debug("Non-retriable exception.", e);
             break;
           }
-          ex = e;
         }
       }
     }
```